File: examples/python/good/lsp-2.py
class Eatable:
    def eat(self, food):
        print(f"I can eat {food}")

class Flyable:
    def fly(self):
        print("Fly, Fly!!")

class Duck(Eatable, Flyable):
    def fly(self):
        print("I quack, quack while flying!!")

class Chicken(Eatable):
  pass

class Plane(Flyable):
    pass

flyables = [Duck(), Plane()]
for each in flyables:
    each.fly()

File: examples/typescript/good/lsp-2.js
"use strict";
class GoodDuck {
    eat(food) {
        console.log(`Duck eats ${food}`);
    }
    fly() {
        console.log("Fly!!!");
    }
}
class Plane {
    fly() {
        console.log("Plane is flying!!!");
    }
}
class GoodChicken {
    eat(food) {
        console.log(`Chicken eats ${food}`);
    }
}
function go_flying(flyables) {
    for (const each of flyables) {
        each.fly();
    }
}
(function () {
    const birds = [new GoodDuck(), new Plane()];
    go_flying(birds);
})();
